core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -87,393 +83,276 @@ public final class LoadManager {
-            throws InterruptedException, AlluxioRuntimeException {
-      if (mCurrentSize.get() >= CAPACITY) {
-        throw new AlluxioRuntimeException(Status.RESOURCE_EXHAUSTED,
-                "Insufficient capacity to enqueue load tasks!", true);
-      }
-
-      mLoadQueue.offer(load);
-      mCurrentSize.incrementAndGet();
+    if (mLoadJobs.values().stream()
+        .map(Load::getStatus)
+        .filter(status -> status != Load.LoadStatus.FAILED && status != Load.LoadStatus.SUCCEEDED)
+        .count() >= CAPACITY) {
+      throw new ResourceExhaustedRuntimeException(
+          "Too many load jobs running, please submit later.");
     }
 
-    void runLoad() {
-      RetryPolicy retryPolicy = new CountingRetry(MAX_RETRY_COUNT);
-
-      while (!Thread.currentThread().isInterrupted()) {
-        try {
-          Load load = mLoadQueue.take();
-          List<FileInfo> listFileInfos = listFileInfos(load);
-          if (listFileInfos == null || listFileInfos.isEmpty()) {
-            LOG.warn(String.format("Get an empty file info list, skip this task %s",
-                    load.mLoadId));
-            mCurrentSize.decrementAndGet();
-            continue;
-          }
-
-          BlockBuffer blockBuffer = BlockBuffer.create(listFileInfos);
-          List<WorkerNetAddress> addresses = getWorkerAddresses();
-
-          addresses.forEach(address -> {
-            while (retryPolicy.attempt()) {
-              try (CloseableResource<BlockWorkerClient> blockWorker
-                           = mFileSystemContext.acquireBlockWorkerClient(address)) {
-                // register worker and set status
-                load.setWorkerLoadFinished(address, false);
-                loadBlockBatch(blockWorker.get(), address, blockBuffer, load);
-              } catch (IOException e) {
-                LOG.warn(String.format("Cannot acquire worker client for %s" + ","
-                        + " after %s tries", address.getHost(), retryPolicy.getAttemptCount()));
+    mLoadJobs.put(load.getPath(), load);
+    mLoadTasks.put(load, new HashMap<>());
+    return true;
+  }
 
-                if (retryPolicy.getAttemptCount() == MAX_RETRY_COUNT) {
-                  addresses.remove(address);
-                  LOG.warn("Remove this worker from the current"
-                          + " worker list after reaching maximum retry count = " + MAX_RETRY_COUNT);
-                }
-              }
-            }
-          });
+  /**
+   * Get active workers.
+   * @return active workers
+   */
+  @VisibleForTesting
+  public Map<WorkerInfo, CloseableResource<BlockWorkerClient>> getActiveWorkers() {
+    return mActiveWorkers;
+  }
 
-          /*If no workers can be acquired, fail the entire load task immediately.*/
-          if (addresses.isEmpty()) {
-            load.addUfsLoadStatus(load.getPath(), TaskStatus.FAILURE);
-          }
+  /**
+   * Refresh active workers.
+   */
+  @VisibleForTesting
+  public void updateWorkers() {
```

Comment:
The underlying implementation of `mFileSystemMaster.getWorkerInfoList()` is already cached and refreshed every `PropertyKey.MASTER_WORKER_INFO_CACHE_REFRESH_TIME` time, which is why we run update in the same frequency here. Though it might worth having a common util to prepare all the clients.

Reply:
yeah I've seen many similar functions, worth a refactoring